### DISCUSSION TITLE
support for JSON Schema?

### DISCUSSION THREAD

**Discussion by wivaku** [spatie/laravel-data#429] [Ideas]
Are there plans for [JSON Schema](https://json-schema.org) support?

I can imagine:
- **generate DTO models** by importing a JSON Schema (similar to https://github.com/wol-soft/php-json-schema-model-generator)
- **export DTO models** to JSON Schema (a la https://spatie.be/docs/laravel-data/v3/advanced-usage/typescript)
  - similar to Typescript → JSON Schema https://github.com/YousefED/typescript-json-schema

**Comment by rubenvanassche** [comment]
I don't think so, this would take us a bit too far from the main objective of the package. You're always welcome to implement such a thing, I'll happily link to it!

**Reply by luckydonald** [reply]
I believe **export** DTO models is as in-scope as the typescript support is.

**Reply by akimkelar** [reply]
It would be really useful and make work easier when using DTO as a JsonResult together with swagger to avoid responses description and just provide the reference to the DTO Schema description.

**Reply by oeholmen** [reply]
I love the idea, and was looking for something like this. Have you done anything on this @wivaku ?

**Reply by wivaku** [reply]
I’m afraid creating that is beyond my skill set

**Comment by BasilLangevin** [comment]
For those still looking for this, I've created a package that generates JSON Schemas from data classes (including validation attributes):

[BasilLangevin/laravel-data-json-schemas](https://github.com/BasilLangevin/laravel-data-json-schemas)
